storage/src/storage/async_adapter.rs

```diff
@@ -0,0 +1,385 @@
+            Message::Canon() => write!(f, "Canon()"),
+            Message::LongestChildPath(hash) => write!(f, "LongestChildPath({})", hash),
+            Message::GetBlockChildren(hash) => write!(f, "GetBlockChildren({})", hash),
+            Message::GetBlockLocatorHashes(canon_depth_limit, oldest_fork_threshold) => write!(
+                f,
+                "GetBlockLocatorHashes({:?}, {})",
+                canon_depth_limit, oldest_fork_threshold
+            ),
+            Message::FindSyncBlocks(hashes, max_block_count) => {
+                write!(f, "FindSyncBlocks(")?;
+                for hash in hashes {
+                    write!(f, "{}, ", hash)?;
+                }
+                write!(f, "{})", max_block_count)
+            }
+            Message::GetTransactionLocation(hash) => write!(f, "GetTransactionLocation({})", hash),
+            Message::GetTransaction(hash) => write!(f, "GetTransaction({})", hash),
+            Message::GetRecordCommitments(limit) => write!(f, "GetRecordCommitments({:?})", limit),
+            Message::GetRecord(hash) => write!(f, "GetRecord({})", hash),
+            Message::StoreRecords(records) => {
+                write!(f, "StoreRecords(")?;
+                for record in records {
+                    write!(f, "{}, ", record.commitment)?;
+                }
+                write!(f, ")")
+            }
+            Message::GetCommitments() => write!(f, "GetCommitments()"),
+            Message::GetSerialNumbers() => write!(f, "GetSerialNumbers()"),
+            Message::GetMemos() => write!(f, "GetMemos()"),
+            Message::GetLedgerDigests() => write!(f, "GetLedgerDigests()"),
+            Message::ResetLedger(_, _, _, _) => write!(f, "ResetLedger(..)"),
+            Message::GetCanonBlocks(limit) => write!(f, "GetCanonBlocks({:?})", limit),
+            Message::GetBlockHashes(limit, filter) => write!(f, "GetBlockHashes({:?}, {:?})", limit, filter),
+            Message::Validate(limit, fix_mode) => write!(f, "Validate({:?}, {:?})", limit, fix_mode),
+            Message::StoreInitDigest(digest) => write!(f, "StoreInitDigest({})", digest),
+            #[cfg(feature = "test")]
+            Message::StoreItem(col, key, value) => write!(f, "StoreItem({:?}, {:?}, {:?})", col, key, value),
+            #[cfg(feature = "test")]
+            Message::DeleteItem(col, key) => write!(f, "DeleteItem({:?}, {:?})", col, key),
+        }
+    }
+}
+
+pub(super) struct Agent<S: SyncStorage + 'static> {
+    inner: S, // the option is only for the purposes of validation, which requires ownership
+}
+
+impl<S: SyncStorage + 'static> Agent<S> {
+    pub(super) fn new(inner: S) -> Self {
+        Agent { inner }
+    }
+
+    fn wrap<T, F: FnOnce(&mut S) -> Result<T>>(&mut self, func: F) -> Result<T> {
+        self.inner.transact(func)
+    }
+
+    fn handle_message(&mut self, message: Message) -> Box<dyn Any + Send + Sync> {
+        trace!("received message: {}", message);
```

Comment:
```suggestion
        trace!("received storage request: {}", message);
```
to differentiate this from the network messages